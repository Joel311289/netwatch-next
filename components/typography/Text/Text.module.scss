$lh-base: 24px;

.wrapper {
  display: block;

  &.xs {
    $size-px: 12px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.sm {
    $size-px: 14px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.md {
    $size-px: 16px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.lg {
    $size-px: 18px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.xl {
    $size-px: 20px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.\32xl {
    $size-px: 24px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.\33xl {
    $size-px: 30px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.\34xl {
    $size-px: 36px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }

  &.\35xl {
    $size-px: 48px;
    @include font(
      $size: $size-px,
      $lh: (
        18px / $size-px
      )
    );
  }
}

.ellipsis {
  @include ellipsis;
}

.maxLines {
  @include max-lines;
}
