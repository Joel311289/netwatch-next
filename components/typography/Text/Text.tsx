import { ElementChildren, ElementHTML, ElementSkeleton } from '@/types';
import { withSkeleton } from '@/hoc/withSkeleton';
import { classes } from '@/utils/helpers';
import styles from '@/components/typography/Text/Text.module.scss';

type TextSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl';

export type TextProps = Partial<typeof defaultProps> &
  ElementHTML &
  ElementChildren<string> &
  ElementSkeleton & {
    size?: TextSize;
    ellipsis?: boolean;
    maxLines?: number;
    truncateTo?: number;
  };

const defaultProps = {
  size: 'md'
};

const Text = ({ className, style, children, size, ellipsis, maxLines, truncateTo }: TextProps) => {
  return (
    <span
      className={classes(
        styles.wrapper,
        className,
        size && styles[size],
        ellipsis && styles.ellipsis,
        maxLines && styles.maxLines
      )}
      style={style}>
      {truncateTo ? children && children.truncate(truncateTo) : children}

      <style jsx>{`
        .${styles.wrapper} {
          ${maxLines ? `-webkit-line-clamp: ${maxLines};` : ''}
        }
      `}</style>
    </span>
  );
};

Text.defaultProps = defaultProps;

const TextWithSkeleton = withSkeleton(Text, { variant: 'text' });

export default TextWithSkeleton;
