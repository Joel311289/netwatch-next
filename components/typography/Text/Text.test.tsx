import { render, screen } from '@testing-library/react';
import Text from '@/components/typography/Text/Text';

describe('Tests Text component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Text>Test</Text>);

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(<Text>Example</Text>);

    expect(screen.getByText('Example')).toBeInTheDocument();
  });

  test('renders truncate correctly', () => {
    const message =
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod, nisi eget consectetur porta, nisl nisi aliquam eros, eget tincidunt nisl nisi eget.';

    render(<Text truncateTo={10}>{message}</Text>);

    expect(screen.getByText(message.truncate(10))).toBeInTheDocument();
  });
});
