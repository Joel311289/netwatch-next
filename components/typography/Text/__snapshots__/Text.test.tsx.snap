// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Text component renders component correctly 1`] = `
<div>
  <span
    class="wrapper md"
  >
    Test
    <style>
      
        .wrapper {
          
        }
      
    </style>
  </span>
</div>
`;
